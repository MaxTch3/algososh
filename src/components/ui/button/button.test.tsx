import { fireEvent, render, screen } from "@testing-library/react"
import { Button } from "./button"

describe('Тест компонента Button', () => {
   it('Отрисовка с текстом', () => {
      const button = render(<Button text='Кнопка' />).container;
      expect(button).toMatchSnapshot();
   });

   it('Отрисовка без текста', () => {
      const button = render(<Button />).container;
      expect(button).toMatchSnapshot();
   });

   it('Отрисовка c блокировкой', () => {
      const button = render(<Button disabled />).container;
      expect(button).toMatchSnapshot();
   });

   it('Отрисовка с индикацией загрузки', () => {
      const button = render(<Button isLoader />).container;
      expect(button).toMatchSnapshot();
   });

   it('Вызов колбека при клике на кнопку', () => {
      const handleClick = jest.fn();
      render(<Button onClick={handleClick} text='Кнопка'/>);
      const button = screen.getByRole('button');
      fireEvent.click(button);
      expect(handleClick).toHaveBeenCalledTimes(1);
   });
})