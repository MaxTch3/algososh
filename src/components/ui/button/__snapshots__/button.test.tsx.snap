// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента Button Отрисовка c блокировкой 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    disabled=""
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Тест компонента Button Отрисовка без текста 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    />
  </button>
</div>
`;

exports[`Тест компонента Button Отрисовка с индикацией загрузки 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined loader "
    disabled=""
    type="button"
  >
    <img
      alt="Загрузка."
      class="loader_icon"
      src="loader.svg"
    />
  </button>
</div>
`;

exports[`Тест компонента Button Отрисовка с текстом 1`] = `
<div>
  <button
    class="text text_type_button text_color_primary button undefined false "
    type="button"
  >
    <p
      class="text undefined"
    >
      Кнопка
    </p>
  </button>
</div>
`;
